library/src/test/java/com/a99/rxplaces/ExtensionsKtTest.kt

```diff
@@ -0,0 +1,62 @@
+package com.a99.rxplaces
+
+import com.a99.rxplaces.options.AutocompleteType
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.CoreMatchers.not
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Test
+import java.util.Random
+
+class ExtensionsKtTest {
+  @Test
+  fun formatComponents_emptyList() {
+    // given
+    val components = emptyList<String>()
+
+    // when
+    val formatted = components.format()
+
+    // then
+    assert(formatted == null)
+  }
+
+  @Test
+  fun formatComponents_oneElement() {
+    // given
+    val components = listOf("country:br")
+
+    // when
+    val formatted = components.format()
+
+    // then
+    assertThat(formatted, `is`("country:br"))
+  }
+
+  @Test
+  fun formatComponents_twoElement() {
+    // given
+    val components = listOf(AutocompleteType.CITIES, AutocompleteType.ESTABLISHMENT)
+
+    // when
+    val formatted = components.format()
+
+    // then
+    assertThat(formatted, `is`("(cities)|establishment"))
+  }
+
+  @Test
+  fun formatComponents_randomElements() {
+    // given
+    val limit = Random().nextInt(10000)
+    val range = 1..limit
+    val components = range.map { "country:$it" }
+
+    // when
+    val formatted = components.format()
+
+    // then
+    assertThat(formatted!!.last().toString(), not(`is`("|")))
```

Comment:
My function is nullable so in that specific scenario I want to throw an exception if my function returns null.